### PR TITLE
Add necessary features to optimizer

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/Cargo.toml [hunk 1]
```diff
@@ -41,21 +41,21 @@ path = "src/lib.rs"
...
 force_hash_collisions = []
 # Used to enable JIT code generation
 jit = ["datafusion-jit", "datafusion-row/jit"]
 pyarrow = ["datafusion-common/pyarrow"]
-regex_expressions = ["datafusion-physical-expr/regex_expressions"]
+regex_expressions = ["datafusion-physical-expr/regex_expressions", "datafusion-optimizer/regex_expressions"]
 # Used to enable scheduler
 scheduler = ["rayon"]
 simd = ["arrow/simd"]
-unicode_expressions = ["datafusion-physical-expr/regex_expressions", "datafusion-sql/unicode_expressions"]
```
Comment (viirya) [inline datafusion/core/Cargo.toml:58]: This was a bug (typo?) actually.

File: datafusion/optimizer/Cargo.toml [hunk 2]
```diff
@@ -33,16 +33,19 @@ name = "datafusion_optimizer"
 path = "src/lib.rs"
 
 [features]
-default = ["unicode_expressions"]
-unicode_expressions = []
+default = ["unicode_expressions", "crypto_expressions", "regex_expressions"]
+crypto_expressions = ["datafusion-physical-expr/crypto_expressions"]
+regex_expressions = ["datafusion-physical-expr/regex_expressions"]
+unicode_expressions = ["datafusion-physical-expr/unicode_expressions"]
+dictionary_expressions = ["datafusion-physical-expr/dictionary_expressions"]
```
Comment (viirya) [inline datafusion/optimizer/Cargo.toml:40]: Otherwise, datafusion-optimizer always pulls in these features from datafusion-physical-expr.

Comment (viirya) [reply datafusion/optimizer/Cargo.toml:40]: For example, we cannot exclude `crypto_expressions` from it.

**Comment by viirya** [comment]
cc @sunchao

**Review by sunchao — APPROVED** [review]
LGTM

**Review by jackwener — APPROVED** [review]

**Pull request merged**